### PR TITLE
Feature Prevent Spam Button When Click Screen Login + register + Forgot Password

### PR DESCRIPTION
**Pull request by tuanlvt-1117**
Task : Feature Prevent Spam Button When Click
Link redmine: https://dev.framgia.com/redmine/issues/151747

### CODE REVIEW

File: app/src/main/java/com/framgia/englishconversation/utils/Constant.java [hunk 1]
```diff
@@ -23,8 +23,9 @@
 
         public static final String POST = "post";
     }
-    
+
     public static final String USER_AGENT = "user-agent";
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/utils/Constant.java]: Please put Constant field to the top of class

File: app/src/main/java/com/framgia/englishconversation/screen/login/LoginViewModel.java [hunk 2]
```diff
@@ -182,4 +197,8 @@ public void setRememberAccount(boolean rememberAccount) {
         mIsRememberAccount = rememberAccount;
         notifyPropertyChanged(BR.rememberAccount);
     }
+
+    private boolean setTimeButtonClick() {
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/login/LoginViewModel.java]: rename set -> check make code look better

**Review by fs-sournary — APPROVED** [review]

**Review by anhpham2403 — APPROVED** [review]

**Review by TheVinhLuong — APPROVED** [review]

**Review by DoanVanToan — APPROVED** [review]

File: app/src/main/java/com/framgia/englishconversation/utils/binding/BindingUtils.java [hunk 3]
```diff
@@ -129,10 +129,22 @@ public static void setDate(TextView view, long milisecond) {
...
     }
+
+    @BindingAdapter({ "onClickSafely" })
+    public static void setOnClickSafely(View view, final View.OnClickListener listener) {
+        view.setOnClickListener(new View.OnClickListener() {
+            private Blocker mBlocker = new Blocker();
+
+            @Override
+            public void onClick(View v) {
+                if (!mBlocker.block()) listener.onClick(v);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/utils/binding/BindingUtils.java:146]: WARNING: 'if' construct must use '{}'s.

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/utils/binding/BindingUtils.java:146]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

**Pull request merged**